jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2423,14 +2423,19 @@ public boolean equalsIgnoreCase(String string) {
 		int currentLength = lengthInternal();
 		
 		/*[IF Sidecar19-SE]*/
+		byte[] buffer = value;
 		// Check if the String is compressed
 		if (enableCompression && count >= 0) {
-			byte[] buffer = new byte[currentLength];
-			compressedArrayCopy(value, 0, buffer, 0, currentLength);
+			if (buffer.length != currentLength) {
+				buffer = new byte[currentLength];
+				compressedArrayCopy(value, 0, buffer, 0, currentLength);
+			}
 			return StringCoding.encode(String.LATIN1, buffer);
 		} else {
-			byte[] buffer = new byte[currentLength * 2];
-			decompressedArrayCopy(value, 0, buffer, 0, currentLength);
+			if (buffer.length != currentLength << 1) {
```

Comment:
`currentLength << 1` is this actually worth doing in bytecode?  I would have expected the JIT to convert the obvious clear code into the right code.

Reply:
Most of the string implementation is using the << operator and it is prevalent in the StringCoding bytecode so I was sticking with that convention. You are right that the JIT would pick which ever is better for the platform in most cases.